Develop

Sammy226 opened the pull request.

Sammy226 committed:
created first draft

created three dice variables and a total number variable which adds up their values

Sammy226 committed:
die class update

re-worked the die class, now it has a dicevalue property and a roll method

Sammy226 committed:
created three dice objects in game class

Sammy226 committed:
Game class updated

Game class now reworked and added a re-roll option

Sammy226 committed:
removed "ReRoll" and added its contents to "RollDice"

Sammy226 committed:
reverted previous changes

Sammy226 committed:
Changed game class again removing ReRoll and started the Program class

issue with the Die class, currently all dice return the same value

Sammy226 committed:
Dice class update

the three dice objects now roll different values however some numbers are out of the range 1 - 6

Sammy226 committed:
Finally fixed the Die  class returning the same number

added the static keyword to the Random method

Sammy226 committed:
Finished the Testing Class and cleaned up code

Testing class has been finished and prevents the dice from being to big / small.

The rest of the code has been cleaned up a bit

JChambers13 reviewed (commented):
I am going to do this review by covering the four questions from the Brief:
1: Have the requirements been met?
I think that the requirements have been met. The code allows the die to be rolled and have the sum outputted. However, there doesn't appear to be much encapsulation used throughout your code. This could be implemented in the die class for the die value. This would mean that the user cannot directly modify the die value. You have included the ability for the dice rolls to be continuous which is good. However, one thing that I would recommend is increasing the use of comments throughout your code, especially for methods. 
This isn't really related to the first question, however I believe it would be beneficial for your game loop to run from the program class instead of the game class. (I could be wrong about this however). This would make your testing easier.
2: Is the code formatted using the style guides correctly?
Your code appears to be following the C# style guides. The only exception is the location of your opening braces, these should be on the same line as the statement and not the next line.
3: Is the code easy to read?
Overall, I think the code is easy to read. Variable names are clear, and the code is not confusing. The one thing that stuck out to me however was the inclusion of 3 roll methods in the die class. I would guess this is due to having issues with the die rolls not being random when using the same method? This is a problem that I had as well. If you make the random object a static, this removes the problem as well. I would recommend changing this as it would make your code shorter and more readable. 
4: Have different errors been handled correctly?
The testing class is not called in the program class. I would recommend calling the test methods to ensure that values are as expected. Furthermore, the game class should be tested as well. Also, I would recommend the use of defensive programming in your loop. This could be in the form of not causing an error message upon wrong input, but simply allowing the user to enter an input again. 
Finally, I would rework the usage of Debug.Assert(). This functions in a way such that a condition is placed in the brackets, followed by a statement that is outputted if the condition is false. For example Debug.Assert(1 ==2 , "This is incorrect") will print the string "This is incorrect". I would recommend using this for your game/die testing.

Overall, your code shows a good understanding of OOP principles/design and I believe it can be improved with a few small changes, such as additional commenting and a rework for testing.

JakeH112233 reviewed (commented):
1: Have the requirements been met?
I would say all of the key points have been met, as the code functionality is all clear and easy to understand. I would say that in future if more comments could be applied to various regions of the code it would also mean that anyone who maybe is not familiar with the language would also be able to understand what the code does.

2: Is the code formatted using the style guides correctly?
Code is written in C# and in the correct program files, so yes!

3: Is the code easy to read?
The code is easy to read and I am able to identify key regions of the code not having to search through to find different parts of the code. The variable names seem all correct, other then that everything else seems in check.

4: Have different errors been handled correctly?
I would say that when the user miss inputs or puts a value that is not correct instead of an error message being shown it should simply allow the user to try and enter a correct value which can be done in the form of a loop of some sort in order to allow them to keep going until they get the correct format of input.  I also think that you should call the testing methods just to ensure that all the values that are there are correct.

In conclusion:
I think that the overall code clearly shows a good level of OOP and can be improved with a few tweaks, with the addition of some clear comments for other people who might not understand the code.

Sammy226 committed:
improved based on peer assessment

i have added comment to the methods

i have used encapsulation on the dice value in the die class

i have moved the re-roll method to the program class
